src/main/java/com/kobylynskyi/graphql/codegen/model/DefinitionTypeDeterminer.java

```diff
@@ -6,27 +6,35 @@
 
 public class DefinitionTypeDeterminer {
 
-    public static GraphqlDefinitionType determine(@NonNull Definition definition) {
-        if (definition instanceof ObjectTypeDefinition) {
+    /**
+     * Determine the type of GraphQL definition.
+     * Comparing using <code>.getClass()</code> in order to ignore subclasses (ExtensionDefinitions in our case)
+     *
+     * @param definition GraphQL definition
+     * @return one of {@link GraphqlDefinitionType}
+     */
+    public static GraphqlDefinitionType determine(@NonNull Definition<?> definition) {
+        Class<?> definitionClass = definition.getClass();
+        if (definitionClass.equals(ObjectTypeDefinition.class)) {
             ObjectTypeDefinition typeDef = (ObjectTypeDefinition) definition;
             if (Utils.isGraphqlOperation(typeDef.getName())) {
                 return GraphqlDefinitionType.OPERATION;
             } else {
                 return GraphqlDefinitionType.TYPE;
             }
-        } else if (definition instanceof EnumTypeDefinition) {
+        } else if (definitionClass.equals(EnumTypeDefinition.class)) {
             return GraphqlDefinitionType.ENUM;
-        } else if (definition instanceof InputObjectTypeDefinition) {
+        } else if (definitionClass.equals(InputObjectTypeDefinition.class)) {
             return GraphqlDefinitionType.INPUT;
-        } else if (definition instanceof SchemaDefinition) {
+        } else if (definitionClass.equals(SchemaDefinition.class)) {
             return GraphqlDefinitionType.SCHEMA;
-        } else if (definition instanceof UnionTypeDefinition) {
+        } else if (definitionClass.equals(UnionTypeDefinition.class)) {
             return GraphqlDefinitionType.UNION;
-        } else if (definition instanceof ScalarTypeDefinition) {
+        } else if (definitionClass.equals(ScalarTypeDefinition.class)) {
             return GraphqlDefinitionType.SCALAR;
-        } else if (definition instanceof InterfaceTypeDefinition) {
+        } else if (definitionClass.equals(InterfaceTypeDefinition.class)) {
             return GraphqlDefinitionType.INTERFACE;
-        } else if (definition instanceof DirectiveDefinition) {
+        } else if (definitionClass.equals(DirectiveDefinition.class)) {
```

Comment:
Rather than this big `if-else` sequence with class checks, why not use directly the `Document.getDeclarationsOfType(...)` to get all declarations of a given class?
We wouldn't have to *then* check which kind of declaration it is, we would just ask for the ones we want at each step.

Reply:
It will slow down the generation process.
Because if you look at ‘getDeclarationsOfType’ method, they iterate through all definitions.
So I think our approach is a liiittle better from performance point of view.

Reply:
Fair enough. For huge schemas this might have some significant overhead.